app/controllers/tags_controller.rb

```diff
@@ -0,0 +1,74 @@
+class TagsController < ApplicationController
+  before_action :logged_in_user, :load_board, :load_list,
+                :check_list_in_board,
+                :check_permission, only: :create
+
+  def create
+    @tag = @list.tags.build tag_params.merge(position: position)
+
+    if @tag.save
+      flash[:success] = t ".success"
+    else
+      flash[:danger] = t ".failed"
+    end
+    redirect_to @board
+  end
+
+  private
+
+  def tag_params
+    params.require(:tag).permit Tag::TAG_PARAMS
+  end
+
+  def load_board
+    @board = Board.find_by id: params[:board_id]
+    if @board
+      return unless @board.closed
+
+      flash[:danger] = t ".board_closed"
+    else
+      flash[:danger] = t ".board_not_found"
+    end
+
+    redirect_to root_url
+  end
+
+  def load_list
+    @list = List.find_by id: params[:list_id]
+    if @list
```

Comment:
```suggestion
    if @list && @list.closed
```
sao lúc nào cũng  redirect_to @board là sao e ?

Reply:
Việc tạo các tag nằm trong page boards/:board_id mà chị, tất cả vấn đề liên quan đến tạo tag tạm thời em sẽ redirect_to lại board đó.
Task sau em sẽ update Ajax cho việc tạo tag ạ.

Reply:
Chỗ này em không edit lại nha, vì bên dưới em có `return unless @list.closed` để flash rõ trường hợp list đóng với list không tồn tại ạ.